src/Todo/Model.hs

```diff
@@ -7,26 +7,22 @@ module Todo.Model
     , Deadline (..)
     , Tags
     , Todos
-    , TodoDB (..)
-    , addTodo
-    , addTag
-    , deleteTag
-    , editTag
     , defTodo
     , getByState
     , getTopPrio
     , getByTag
     , tagStatistics
     ) where
 
-import           Data.List.NonEmpty (NonEmpty (..), toList, fromList)
+import           Data.List.NonEmpty (NonEmpty (..), toList)
 import qualified Data.Ratio         as R
 import qualified Numeric.Natural    as N
 import qualified Data.List          as List
+import qualified Data.Ord           as Ord
```

Comment:
Actually, you don't need to go full-in with qualified imports. `qualified` should be used to avoid collisions. Well. actually, there're some well-known modules which better be imported qualified. But here it's better to use explicit import lists. This will make import section more helpful and code less noisy.

```haskell
import Data.Ord (comparing)
import Numeric.Natural (Natural)
-- and so on
```

Also, you don't need to reexport `toList` from `NonEmpty` module because `toList` works for any `Foldable`, it's already in `Prelude` (most likely).

Reply:
Actually it does not compile without `toList` being exported.

```
    • Variable not in scope: toList :: Tags -> [Tag]
    • Perhaps you want to add ‘toList’ to the import list
      in the import of ‘Data.List.NonEmpty’ (src/Todo/Model.hs:18:1-53)
```

Reply:
Ah, I see. Okay, looks like `toList` is not in `Prelude` indeed.